src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs

```diff
@@ -61,6 +67,22 @@ await _documentClient.CreateDatabaseIfNotExistsAsync(new Database {Id = _options
                 new RequestOptions {OfferThroughput = _options.Value.OfferThroughput}).ConfigureAwait(false);
         }
 
+        private async Task CreateLeaseRequestsCollection()
+        {
+            var collectionDefinition = new DocumentCollection
+            {
+                Id = _options.Value.RequestsCollection
+            };
+
+            collectionDefinition.PartitionKey.Paths.Add("/leaseType");
+            collectionDefinition.DefaultTimeToLive = 30;
+
```

Comment:
Can we set an unique constraint on (WorkType,InstanceId) on LeaseRequests collection?